plugins/modules/packaging/os/pulp_repo.py

```diff
@@ -341,6 +376,65 @@ def create_repo(
                 "_repo-type": "rpm-repo"
             }
 
+        if self.repo_type == 'deb':
+            deb_distributor = dict()
+            deb_distributor['distributor_id'] = "deb_distributor"
+            deb_distributor['distributor_type_id'] = "deb_distributor"
+            deb_distributor['auto_publish'] = True
+            deb_distributor['distributor_config'] = dict()
+            deb_distributor['distributor_config']['http'] = serve_http
+            deb_distributor['distributor_config']['https'] = serve_https
```

Comment:
There seems to be **a lot** of shared code between the code paths. I think it would make sense to consolidate some of it.